### PR TITLE
Task/46: Clean up UI code - Add reusable `AuthenticedLayout` component and update other pages to consume

### LINKED ISSUE #46: [Chore] DRY UI layouts
## Issue

The layout component is very minimal:

https://github.com/nhantran3395/scrapper/blob/448ebb7f173de08329614fbc14c41673c7099651/apps/web/src/components/common/layout/layout.tsx#L7

As a result, all pages now repeat the code for the layout that includes the navigation bar, the loading notifications, etc.

https://github.com/nhantran3395/scrapper/blob/448ebb7f173de08329614fbc14c41673c7099651/apps/web/src/components/details/details.tsx#L39-L41

https://github.com/nhantran3395/scrapper/blob/448ebb7f173de08329614fbc14c41673c7099651/apps/web/src/components/keywords/keywords.tsx#L49-L51

## Expected

Ensure there is a layout to with all shared components

> [!Note]
> There can be two layout file: one for auth pages and another one for the other pages.

### PR DESCRIPTION
**Pull request by nhantran3395**
#### What does this PR do?

This PR is to clean up UI code by eliminating repeated code related to setting up the layout and Navigation Bar for each page that is authenticated
Close #46 

#### How is it implemented?

- Add new component `AuthenticatedLayout`
- Update pages (`keywords`, `uploads`,` details`) that consume this new `AuthenticatedLayout`

#### Unit tests

Added tests to verify that the new `AutheticatedLayout` and old `Layout` is rendering child components correctly

image

#### Screenshots

The page stays  unchanged after the update:

- `uploads` page
Screenshot 2023-10-09 at 22 14 49

- `keywords` page
Screenshot 2023-10-09 at 22 15 00

- `details` page
Screenshot 2023-10-09 at 22 15 08

### CODE REVIEW

File: apps/web/src/components/common/layout/autheticated-layout.test.tsx
Comment (olivierobert) [inline apps/web/src/components/common/layout/autheticated-layout.test.tsx:1]: Typo in the file name: authe**n**ticated-layout.test.tsx

File: apps/web/src/components/common/layout/authenticated-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export function AuthenticatedLayout({ children }: AuthenticatedLayoutProps) {
+  const { signOut } = useUser({
+    redirectTo: "/login",
+  });
+
+  return (
+    <main className={"min-h-screen bg-slate-50 flex flex-1"}>
+      <div className="h-full w-full">
+        <NavigationBar signOut={signOut} />
+        <article className="mx-auto max-w-7xl py-6 px-6 sm:px-6 lg:px-8">
```
Comment (olivierobert) [inline apps/web/src/components/common/layout/authenticated-layout.tsx:20]: I was initially unsure about the nesting of `<section/>` instead `<article/>` but it seems to be semantically correct: https://www.w3schools.com/html/html5_semantic_elements.asp#:~:text=Nesting%20%3Carticle%3E%20in%20%3Csection%3E%20or%20Vice%20Versa%3F

**Review by olivierobert — APPROVED** [review]
Just one mini typo. The rest looks good.

File: apps/web/src/components/common/layout/autheticated-layout.test.tsx
Comment (nhantran3395) [reply apps/web/src/components/common/layout/autheticated-layout.test.tsx:1]: Thanks for spotting this! I have updated the PR to fix

File: apps/web/src/components/common/layout/authenticated-layout.tsx (hunk 1 above)
Comment (nhantran3395) [reply apps/web/src/components/common/layout/authenticated-layout.tsx:20]: I don't know until now that it's controversial also 😆 
https://stackoverflow.com/questions/9527378/should-sections-have-articles-or-should-articles-have-sections#:~:text=It's%20entirely%20acceptable%20to%20nest,they%20are%20two%20different%20things.

My thinking when I used this was a page can have one or many articles, and each of the articles can have many sections, so it seems correct to nest `section` in `article`

**Pull request merged**